app/services/events/stores/aggregated_clickhouse_store.rb

```diff
@@ -390,16 +394,82 @@ def sum_date_breakdown
       end
 
       def weighted_sum(initial_value: 0)
-        # TODO(pre-aggregation): Implement
+        result = connection_with_retry do |connection|
+          query = Events::Stores::AggregatedClickhouse::WeightedSumQuery.new(store: self)
+
+          sql = ActiveRecord::Base.sanitize_sql_for_conditions(
+            [
+              sanitize_colon(query.query),
+              {
+                from_datetime:,
+                to_datetime: to_datetime.ceil,
+                decimal_scale: DECIMAL_SCALE,
+                initial_value: initial_value || 0
+              }
+            ]
+          )
+
+          connection.select_one(sql)
+        end
+
+        result["aggregation"]
       end
 
       def grouped_weighted_sum(initial_values: [])
-        # TODO(pre-aggregation): Implement
+        connection_with_retry do |connection|
+          query = AggregatedClickhouse::WeightedSumQuery.new(store: self)
+
+          # NOTE: build the list of initial values for each groups
+          #       from the events in the period
+          formated_initial_values = grouped_count.map do |group|
```

Comment:
I feel like there might be a simpler way to handle this. Something like:

```ruby
groups_in_period = grouped_count.pluck(:groups)
groups_in_initial_values = initial_values.pluck(:groups)
formatted_initial_values = initial_values.dup
(groups_in_period - groups_in_initial_values).each do |groups|
  formatted_initial_values << { groups: , value: 0 }
end
```

But I might be missing something